tests/mocha/block_test.js

```diff
@@ -1238,7 +1238,447 @@ suite('Blocks', function() {
       });
     });
   });
+  suite('Collapsing and Expanding', function() {
+    function assertCollapsed(block, opt_string) {
+      chai.assert.isTrue(block.isCollapsed());
+      for (var i = 0, input; (input = block.inputList[i]); i++) {
+        if (input.name == Blockly.Block.COLLAPSED_INPUT_NAME) {
+          continue;
+        }
+        console.log(input);
+        chai.assert.isFalse(input.isVisible());
+        for (var j = 0, field; (field = input.fieldRow[j]); j++) {
+          chai.assert.isFalse(field.isVisible());
+        }
+      }
+      var icons = block.getIcons();
+      for (var i = 0, icon; (icon = icons[i]); i++) {
+        chai.assert.isFalse(icon.isVisible());
+      }
+
+      var input = block.getInput(Blockly.Block.COLLAPSED_INPUT_NAME);
+      chai.assert.isNotNull(input);
+      chai.assert.isTrue(input.isVisible());
+      var field = block.getField(Blockly.Block.COLLAPSED_FIELD_NAME);
+      chai.assert.isNotNull(field);
+      chai.assert.isTrue(field.isVisible());
+
+      if (opt_string) {
+        chai.assert.equal(field.getText(), opt_string);
+      }
+    }
+    function assertExpanded(block) {
+      chai.assert.isFalse(block.isCollapsed());
+      for (var i = 0, input; (input = block.inputList[i]); i++) {
+        chai.assert.isTrue(input.isVisible());
+        for (var j = 0, field; (field = input.fieldRow[j]); j++) {
+          chai.assert.isTrue(field.isVisible());
+        }
+      }
 
+      var input = block.getInput(Blockly.Block.COLLAPSED_INPUT_NAME);
+      chai.assert.isNull(input);
+      var field = block.getField(Blockly.Block.COLLAPSED_FIELD_NAME);
+      chai.assert.isNull(field);
+    }
+    function assertHidden(block) {
```

Comment:
I changed my mind later, but forgot to update my comment. I can see how `assertVisible` might not be as clear because we are checking the css and not isVisible property.